### PR TITLE
kustomize: Add appropriate permissions for deployment on OCP

### PR DESCRIPTION
**Pull request by swatisehgal**
We add an overlay especially for openshift to ensure that we provide appropriate permissions for successful deployment of this plugin on OCP.

Signed-off-by: Swati Sehgal <swsehgal@redhat.com>

### CODE REVIEW

File: deployment/base/rbac/scc.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# TODO: check if we can reduce scc capabilities
```
Comment (ffromani) [inline deployment/base/rbac/scc.yaml]: can we use one of the default OCP SCCs? If memory serves, we do have a builtin SCC very similar to this

Comment (swatisehgal) [reply deployment/base/rbac/scc.yaml]: Ack, will look into this.

Comment (swatisehgal) [reply deployment/base/rbac/scc.yaml]: We can run it with `privileged` SCC. Updated to that.

Comment (swatisehgal) [reply deployment/base/rbac/scc.yaml]: Here is what privileged SCC looks like:
```
NAME                              PRIV    CAPS         SELINUX     RUNASUSER          FSGROUP     SUPGROUP    PRIORITY     READONLYROOTFS   VOLUMES

privileged                        true    ["*"]        RunAsAny    RunAsAny           RunAsAny    RunAsAny    <no value>   false            ["*"]

```

**Comment by Tal-or** [comment]
Can't we use the same DSs plus some patches? 
I think that having a separate DSs is too much and we're not using the flexibility that Kustomize has to offer.
I'll try to provide an example

**Comment by swatisehgal** [comment]
> Can't we use the same DSs plus some patches? I think that having a separate DSs is too much and we're not using the flexibility that Kustomize has to offer. I'll try to provide an example

I tried that but the problem was that the original device plugin daemon set files were not in or below the overlay and I ran into the following:
```
$ kubectl apply -k deployment/overlays/openshift/
error: accumulating resources: accumulation err='accumulating resources from '../../base/devicepluginA-ds-ocp': '/home/swsehgal/go/src/github.com/k8stopologyawareschedwg/sample-device-plugin/deployment/base/devicepluginA-ds-ocp' must resolve to a file': recursed accumulation of path '/home/swsehgal/go/src/github.com/k8stopologyawareschedwg/sample-device-plugin/deployment/base/devicepluginA-ds-ocp': accumulating resources: accumulation err='accumulating resources from '../devicepluginA-ds/devicepluginA-ds.yaml': security; file '/home/swsehgal/go/src/github.com/k8stopologyawareschedwg/sample-device-plugin/deployment/base/devicepluginA-ds/devicepluginA-ds.yaml' is not in or below '/home/swsehgal/go/src/github.com/k8stopologyawareschedwg/sample-device-plugin/deployment/base/devicepluginA-ds-ocp'': must build at directory: '/home/swsehgal/go/src/github.com/k8stopologyawareschedwg/sample-device-plugin/deployment/base/devicepluginA-ds/devicepluginA-ds.yaml': file is not directory
```

**Comment by swatisehgal** [comment]
I will give this another go, even I don't like having the entire daemonset files in the new overlays just because we need to specify serviceaccount :(

**Comment by swatisehgal** [comment]
Thanks @Tal-or for [PR](https://github.com/k8stopologyawareschedwg/sample-device-plugin/pull/16) with nice fix using `patchesStrategicMerge`!

**Comment by swatisehgal** [comment]
Ah, misclicked!

Reopening

**Review by ffromani — APPROVED** [review]
/approve
thanks!

**Pull request merged**